flake.nix

```diff
@@ -0,0 +1,76 @@
+{
+  description = "Simon’s Improved Layout Engine";
+
+  # To make user overrides of the nixpkgs flake not take effect
+  inputs.nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+  inputs.flake-utils.url = "github:numtide/flake-utils";
+
+  # https://nixos.wiki/wiki/Flakes#Using_flakes_project_from_a_legacy_Nix
+  inputs.flake-compat = {
+    url = "github:edolstra/flake-compat";
+    flake = false;
+  };
+
+  outputs = { self
+    , nixpkgs
+    , flake-utils
+    , flake-compat
+  }:
+  flake-utils.lib.eachDefaultSystem (system:
+    let
+      pkgs = import nixpkgs {
+        inherit system;
+      };
+      # TODO: Should this be replaced with libtexpdf package from nixpkgs? or
+      # should we keep it that way, so that it'd be easy to test new versions
+      # of libtexpdf when developing?
+      libtexpdf-src = builtins.fetchGit {
+        url = "https://github.com/sile-typesetter/libtexpdf";
+        rev = "${(pkgs.lib.fileContents "${self}/libtexpdf.git-rev")}";
+      };
+      # https://discourse.nixos.org/t/passing-git-commit-hash-and-tag-to-build-with-flakes/11355/2
+      version_rev = if (self ? rev) then (builtins.substring 0 8 self.rev) else "dirty";
+      # Use the expression from Nixpkgs instead of rewriting it here.
+      sile = pkgs.sile.overrideAttrs(oldAttr: rec {
+        version = "${(pkgs.lib.importJSON ./package.json).version}-${version_rev}-flake";
+        src = builtins.filterSource
+          (path: type: type != "directory" || baseNameOf path != ".git")
+        ./.;
```

Comment:
> `git clean -dxff -e node_modules -e .fonts -e .sources -e .husky`

Perhaps we should filter these files as well right here?

Reply:
I don't think that will help, if `$CWD` is still set to the project directory when it is actually run it won't matter what was or wasn't copied into the Nix build, SILE will still prefer local resources to it's distribution ones. This is intentional so that users can override virtually anything in the system, but it does make the assumption that any resources in `$CWD` should be usable.

In any case this is a developer only problem and I'm not too worried about it. I've scheduled this for merge so I can test it out on the SILE website before tagging a release. We can issue any adjustments for more developer convenience in a follow-up.

Reply:
As @ctrlcctrlv noted, this is really the same as #1212. I'm not sure what the solution should even be (maybe nothing), but lets move discussion there.